src/hardware/serialport/misc_util.h

```diff
@@ -55,22 +55,28 @@
 
 #include <SDL_net.h>
 
-Bit32u Netwrapper_GetCapabilities();
+uint32_t Netwrapper_GetCapabilities();
 
 struct _TCPsocketX {
-	int ready;
+	int ready = 0;
 #ifdef NATIVESOCKETS
-	SOCKET channel;
+	SOCKET channel = 0;
 #endif
-	IPaddress remoteAddress;
-	IPaddress localAddress;
-	int sflag;
+	IPaddress remoteAddress = {0, 0};
+	IPaddress localAddress = {0, 0};
+	int sflag = 0;
+};
+
+enum class SocketState {
+	Good, // had data and socket is open
+	Empty, // didn't have data but socket is open
+	Closed // didn't have data and socket is closed
```

Comment:
The only really notable addition here.
Everything else is just type-swaps, and replacing `Bitu`'s for `size_t`s (because they index/size an array.